kibana/templates/configmap-helm-scripts.yaml

```diff
@@ -4,18 +4,161 @@ kind: ConfigMap
 metadata:
   name: {{ template "kibana.fullname" . }}-helm-scripts
   labels: {{ include "kibana.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade,post-delete
+    "helm.sh/hook-delete-policy": hook-succeeded
+    {{- if .Values.annotations }}
+    {{- range $key, $value := .Values.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+    {{- end }}
 data:
-  kibana-entrypoint.sh: |
-    #!/bin/bash
-    set -euo pipefail
+  manage-es-token.js: |
+    const https = require('https');
+    const fs = require('fs');
 
-    echo "export ELASTICSEARCH_SERVICEACCOUNTTOKEN=$({{ template "kibana.home_dir" . }}/node/bin/node {{ template "kibana.home_dir" . }}/helm-scripts/parse-token.js {{ template "kibana.home_dir" . }}/config/tokens/{{ template "kibana.fullname" . }}.json)" > $HOME/.elasticsearch-serviceaccounttoken
-    source $HOME/.elasticsearch-serviceaccounttoken
+    // Elasticsearch API
+    const esPath = '_security/service/elastic/kibana/credential/token/{{ template "kibana.fullname" . }}';
+    const esUrl = '{{ .Values.elasticsearchHosts }}' + '/' + esPath
+    const esUsername = process.env.ELASTICSEARCH_USERNAME;
+    const esPassword = process.env.ELASTICSEARCH_PASSWORD;
+    const esAuth = esUsername + ':' + esPassword;
```

Comment:
Should we validate that the username and password have been supplied? I would be better to catch that explicitly if we'd misconfigured or failed to propagate something.

Reply:
`ELASTICSEARCH_USERNAME` and `ELASTICSEARCH_PASSWORD` variables are defined in the Job pod template where this script is supposed to run:

https://github.com/elastic/helm-charts/blob/81ce165618d2b7602bff2e814e9825800c087aa0/kibana/templates/pre-install-job.yaml#L28-L37

If these variables can't be filled (because the K8S secret doesn't exist or doesn't have the expected format), the job will not start and not trigger the script. So theoretically, there is no need to add validation, but I agree, this would be better 👍🏻

I'll try to provide an update for that.

Reply:
ff4a1c60b02b3553120dc97454519a1917e90085